kore/src/Kore/Exec.hs

```diff
@@ -644,11 +651,23 @@ See 'checkEquation',
 -}
 checkFunctions ::
     MonadLog m =>
-    MonadSimplify m =>
+    MonadMask m =>
+    MonadProf m =>
+    MonadIO m =>
+    MonadSMT m =>
     -- | The main module
     VerifiedModule StepperAttributes ->
     m ()
-checkFunctions verifiedModule = do
+checkFunctions verifiedModule = evalSimplifier verifiedModule' $ do
+    -- let equationMap = extractEquations verifiedModule'
+    equationMap <-
+        extractEquations verifiedModule'
+            & (Map.map . map . Equation.mapVariables) (pure mkEquationVariable)
+            & simplifyExtractedEquations
+    let equations =
+            Map.elems equationMap
+                & map (functionRules . partitionEquations)
```

Comment:
`functionRules . partitionEquations` essentially filters out simplification rules and only keeps the function rules.
See: [`partitionEquations`](https://github.com/kframework/kore/blob/78ec7647e9b9eca9045544fe4a8c6c1fead93587/kore/src/Kore/Equation/Registry.hs#L101)